README.md

```diff
@@ -58,3 +58,79 @@ xRequestId := req.Header.Get(athenahealth.XRequestIDHeaderKey))
 ```
 
 See the athena [Best Practices](https://docs.athenahealth.com/api/guides/best-practices) guide for more details about X-Request-Id and other recommended  practices.
+
+## Method Signatures Required vs. Optional Fields
+
+All methods that perform network or filesystem IO will accept a context for idiomatic propagation.
+
+While not yet consistently applied across this repo, required fields should be surfaced as top-level arguments in method signatures (including path parameters, query parameters, and request body fields) while optional fields should be embedded as pointer fields in an optional pointer struct. The aim is to provide a consistent and natural pattern for interacting with this client interface as well as to add compile-time safety to required fields.
+
+### Required and Optional Fields
+
+Optional fields options struct should by the last argument in the method signature, preceded by top-level required fields.
```

Comment:
Do we want to mention here that validation of required fields is recommended to provide faster time to failure and clearer error messaging for those cases? 

For example the pattern being added in https://github.com/eleanorhealth/go-athenahealth/pull/115/files#diff-3728f270059f7f7e670e8321b74c39538272678007f2063be6b85d7176922b54R61-R70